pytorch_lightning/trainer/trainer.py

```diff
@@ -88,6 +88,7 @@ def __init__(
         callbacks: Optional[Union[List[Callback], Callback]] = None,
         default_root_dir: Optional[str] = None,
         gradient_clip_val: float = 0,
+        gradient_clip_algorithm: str = 'norm2',
```

Comment:
I think that using two separate arguments, for example, `gradient_clip_algorithm='norm'` and `gradient_clip_algorithm_norm_type=2`, has much better readability than the current way.